apisix/core/config_etcd.lua

```diff
@@ -188,9 +190,43 @@ local function run_watch(premature)
 
         ::watch_event::
         while true do
-            local res, err = res_func()
-            if log_level >= NGX_INFO then
-                log.info("res_func: ", inspect(res))
+            local res
+            local sema, err = semaphore.new()
+            if not sema then
+                log.error("create sema failed", err)
+                goto watch_event
+            end
+
+            local get_res_th = ngx_thread_spawn(function ()
```

Comment:
@leslie-tsang 
I'm concerned that each watch will spawn two threads, which will affect performance.
What if we check exiting() after the watch has timed out? It's currently 50 seconds, maybe we should lower it?

Reply:
How long do you think is appropriate? 1s?

Reply:
Forget that, it doesn't seem like a good point.